### PR TITLE
Added v3 of Laghos

### PR DESCRIPTION
**Pull request by rspavel**
Added v3 of Laghos as per
https://github.com/CEED/Laghos/blob/v3.0/README.md

### CODE REVIEW

File: var/spack/repos/builtin/packages/laghos/package.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class Laghos(MakefilePackage):
     git      = "https://github.com/CEED/Laghos.git"
 
     version('develop', branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/laghos/package.py:21]:
```suggestion
    version('master', branch='master')
```

File: var/spack/repos/builtin/packages/laghos/package.py [hunk 2]
```diff
@@ -36,6 +37,11 @@ class Laghos(MakefilePackage):
     depends_on('mfem@3.3.1-laghos-v1.0:+mpi+metis', when='@1.0,1.1+metis')
     depends_on('mfem@3.3.1-laghos-v1.0:+mpi~metis', when='@1.0,1.1~metis')
 
+    # 3.0 requirements
+    depends_on('hypre@2.11.2', when='@3.0:')
+    depends_on('mfem@develop+mpi+metis', when='@3.0:+metis')
+    depends_on('mfem@develop+mpi~metis', when='@3.0:~metis')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/laghos/package.py:43]: I think you can remove the following lines above:
```python
depends_on('mfem@develop+mpi+metis', when='@develop+metis')
depends_on('mfem@develop+mpi~metis', when='@develop~metis')
```
`master` and `develop` both sort as higher than `@3.0:`

File: var/spack/repos/builtin/packages/laghos/package.py [hunk 3]
```diff
@@ -46,6 +52,11 @@ def build_targets(self):
         targets.append('TEST_MK=%s' % spec['mfem'].package.test_mk)
         targets.append('CXX=%s' % spec['mpi'].mpicxx)
 
+        if self.version >= ver('3.0'):
+            targets.append('HYPRE_DIR=%s' % spec['hypre'].prefix)
+            if '+metis' in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/laghos/package.py:57]: Shouldn't we add a metis dependency?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/laghos/package.py:57]: We also probably want a conflict for `+metis` when `@:2`

File: var/spack/repos/builtin/packages/laghos/package.py (hunk 2 above)
Comment (rspavel) [reply var/spack/repos/builtin/packages/laghos/package.py:42]: The idea was to eventually figure out their actual requirements regarding mfem and specify that for v3 but with the amount of overhaul required on this there is no reason to duplicate logic now.

Fixed

File: var/spack/repos/builtin/packages/laghos/package.py (hunk 3 above)
Comment (rspavel) [reply var/spack/repos/builtin/packages/laghos/package.py:56]: Made metis dependency explicit.

To my understanding, ```@:2``` still depends on metis but in ways that are much less defined

**Pull request merged**